tests/main/snapctl-is-connected-pid/task.yaml

```diff
@@ -1,68 +1,86 @@
 summary: Ensure that "snapctl is-connected --pid" works.
 
 prepare: |
-  #shellcheck source=tests/lib/snaps.sh
-  . "$TESTSLIB"/snaps.sh
-  install_local test-snap1
-  install_local test-snap2
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snap1
+    install_local test-snap2
+
+    #shellcheck source=tests/lib/dirs.sh
+    . "$TESTSLIB"/dirs.sh
+    case "$SPREAD_SYSTEM" in
+        fedora-*|arch-*|centos-*)
+            # although classic snaps do not work out of the box on fedora,
+            # we still want to verify if the basics do work if the user
+            # symlinks /snap to $SNAP_MOUNT_DIR themselves
+            ln -sf $SNAP_MOUNT_DIR /snap
```

Comment:
You can use `tests.cleanup prepare` at the top of `prepare`, then `tests.cleanup restore` in `restore`. This will let you run `tests.clenaup defer rm -f /snap` without repeating the condition in restore. Just do it here, it's nice for cleanup logic, as it's right next to the code that modifies the system.